doc/manual/noteworthy-differences.rst

```diff
@@ -251,3 +251,94 @@ Noteworthy differences from Python
+  will automatically be promoted to a larger size type, such as ``Int64`` (if ``Int`` is ``Int32``),
+  ``Int128``, or the arbitrarily large ``BigInt`` type.
+  There are no numeric literal suffixes, such as ``L``, ``LL``, ``U``, ``UL``, ``ULL`` to indicate unsigned
+  and/or signed vs. unsigned.
+  Decimal literals are always signed, and hexadecimal literals (which start with ``0x`` like C/C++),
+  are unsigned (except if they are >128 bits, in which case they become signed ``BigInt`` type).
+  Hexadecimal literals also, unlike C/C++/Java and unlike decimal literals in Julia,
+  have a type based on the *length* of the literal, including leading 0s.  For example,
+  ``0x0`` and ``0x00`` have type UInt8, ``0x000`` and ``0x0000`` have type ``UInt16``, then
+  literals with 5 to 8 hex digits have type ``UInt32``, 9 to 16 hex digits type ``UInt64``, and more than
+  16 hex digits end up a signed ``BigInt`` type.  This needs to be taken into account when defining
+  hexadecimal masks, for example ``~0xf`` is very different from ``~0x000f``.
+  Unlike either decimal integer or hexadecimal literals, floating point literals are always
+  64-bits, ``Float64``, and don't get promoted to the ``BigFloat`` type.  This is closer in
+  behavior to C/C++.
+  Octal (prefixed with ``0o``) and binary (prefixed with ``0b``) literals are also treated as unsigned
+  (with the same current exception that they become signed ``BigInt`` type if they can't fit in a ``UInt128``).
+- String literals can be delimited with either ``"``  or ``"""``, ``"""`` delimited literals can contain ``"``
+  characters without quoting it like ``"\\\""``
+  String literals can have values of other variables or expressions interpolated into them,
+  indicated by ``$variablename`` or ``$(expression)``, which evaluates the variable name or the expression in the context of the function.
+- ``//`` indicates a ``Rational`` number, and not a single-line comment (which is # in Julia)
+- ``#=`` indicates the start of a multiline comment, and ``=#`` ends it.
+- Functions in Julia return values from their last expression(s) or the ``return``
+  keyword.  Multiple values can be returned from functions and assigned as tuples, e.g.
+  ``(a, b) = myfunction()`` or ``a, b = myfunction()``, instead of having to pass pointers
+  to values as one would have to do in C/C++ (i.e. ``a = myfunction(&b)``.
+- Julia discourages the used of semicolons to end statements. The results of
+  statements are not automatically printed (except at the interactive prompt, i.e. the REPL),
+  and lines of code do not need to end with semicolons. :func:`println` or
+  :func:`@printf` can be used to print specific output.
+  In the REPL, ``;`` can be used to suppress output.
+  ``;`` also has a different meaning within ``[ ]``, something to watch out for.
+  ``;`` can be used to separate 'statements' on a line, but are not strictly necessary in many cases,
```

Comment:
Ok, good... I put it in quotes because I knew it wasn't *really* a statement... maybe it should also point out as you said that Julia is expression-oriented, not statement oriented as C/C++.